# Competition configuration options
options:
  insert_models_over_bins: true
  insert_models_over_shelves: true
  belt_population_cycles: 1

shelf_layout:
  row_1: ['base',0,'collar','base'] #shelves: 3, 4, 5
  row_2: [0, 'pipe','base','base']   #shelves: 6, 7, 8
  row_3: ['base','pipe',0,'collar'] #shelves: 9, 10, 11

aisle_layout:
  person_1:
    location: -1.57
    start_time: 0.
    move_time: 9.
    wait_time: 7.
  person_2:
    location: 4.57
    start_time: 16.
    move_time: 9.
    wait_time: 7.

orders:
  order_0:
    announcement_condition: time
    announcement_condition_value: 0.0
    shipment_count: 2
    destinations: [agv1, agv2]
    products:
      part_0:
        type: piston_rod_part_blue
        pose:
          xyz: [0.1, -0.1, 0]
          rpy: [0, 0, 0]
      part_1:
        type: gear_part_green
        pose:
          xyz: [-0.1, 0, 0]
          rpy: [0, 0, 'pi']
      part_2:
        type: gear_part_red
        pose:
          xyz: [-0.1, 0.15, 0]
          rpy: [0, 0, 0]

models_over_bins:
   bin3:
     models:
       gear_part_green:
         xyz_start: [0.2, 0.2, 0.0]
         xyz_end: [0.4, 0.4, 0.0]
         rpy: [0, 0, 0]
         num_models_x: 2
         num_models_y: 2
   bin4:
     models:
       piston_rod_part_blue:
         xyz_start: [0.1, 0.2, 0.0]
         xyz_end: [0.4, 0.5, 0.0]
         rpy: [0, 0, 'pi/4']
         num_models_x: 2
         num_models_y: 3

models_over_shelves:
  shelf1:
    models:
      gear_part_red:
        xyz_start: [-1.5, -0.5, 0.0]
        xyz_end: [-1.0, -0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 5
        num_models_y: 1
  shelf5:
    models:
      gear_part_red:
        xyz_start: [-1.5, 0.5, 0.0]
        xyz_end: [-1.0, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 5
        num_models_y: 1
  shelf8:
    models:
      gear_part_green:
        xyz_start: [0.7, -0.5, 0.0]
        xyz_end: [0., -0.4, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 5
        num_models_y: 2
  shelf11:
    models:
      piston_rod_part_blue:
        xyz_start: [1.5, 0.4, 0.0]
        xyz_end: [1.0, 0.4, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 3
        num_models_y: 1
      piston_rod_part_green:
        xyz_start: [1.5, -0.5, 0.0]
        xyz_end: [1.0, -0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 3
        num_models_y: 1

belt_models:
  piston_rod_part_blue:
    1.0:
      pose:
        xyz: [0.0, 4.3, 0.92]
        rpy: [0, 0, 'pi/2']
  gear_part_red:
    2.0:
      pose:
        xyz: [0.0, 4.3, 0.92]
        rpy: [0, 0, 'pi/2']

time_limit: -1
